action.yml

```diff
@@ -1,36 +1,119 @@
+
+    - name: Set Tutor environment variables
+      run: |
+        echo "LMS_HOST=local.edly.io" >> "$GITHUB_ENV"
+        echo "CMS_HOST=studio.local.edly.io" >> "$GITHUB_ENV"
+        echo "TUTOR_ROOT=$(pwd)" >> "$GITHUB_ENV"
+        echo "TUTOR_PLUGINS_ROOT=$(pwd)/plugins/" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Install and prepare Tutor
+      run: |
+        pip install "tutor${{ inputs.tutor_version }}"
+        tutor config save --set LMS_HOST=$LMS_HOST --set CMS_HOST=$CMS_HOST
+        chmod 777 . -R
+        tutor local launch -I
+      shell: bash
+
+    - name: Configure Caddyfile and Open edX settings
+      run: |
+        mkdir -p plugins
+        cat << 'EOF' > plugins/patches.yml
+
+        name: patches
+        patches:
+          caddyfile: |
+            {$default_site_port} {
+              import proxy "lms:8000"
+            }
+          openedx-cms-production-settings: |
+            ALLOWED_HOSTS = ["*"]
+            ALLOWED_AUTH_APPLICATIONS = ['cms-sso', 'cms-sso-dev']
+          openedx-lms-production-settings: |
+            ALLOWED_HOSTS = ["*"]
+            ALLOWED_AUTH_APPLICATIONS = ['cms-sso', 'cms-sso-dev']
+        EOF
+        tutor plugins enable patches
+      shell: bash
+
+    - name: Install plugin and extra requirements
+      run: |
+        tutor local exec lms pip install ${{ inputs.app_name }} ${{ inputs.openedx_extra_pip_requeriments }}
+        tutor local exec cms pip install ${{ inputs.app_name }} ${{ inputs.openedx_extra_pip_requeriments }}
+      shell: bash
+
+    - name: Run migrations
+      run: |
+        tutor local exec lms python manage.py lms migrate
+        tutor local exec cms python manage.py cms migrate
+        tutor local restart
+      shell: bash
+
+    - name: Load inital data for the tests
+      if: ${{ inputs.fixtures_file }}
+      run: |
+        echo "Copying fixtures file to the LMS container"
+        tutor local exec lms mkdir -p /openedx/fixtures
+        tutor local dc cp ${{ inputs.fixtures_file }} lms:/openedx/fixtures/fixture.json
+        tutor local exec lms python manage.py lms loaddata /openedx/fixtures/fixture.json
+      shell: bash
+
+    - name: Curl Heartbeat
+      run: |
+        echo "Curling LMS heartbeat"
+        status_code=$(curl -s -o /dev/null -w "%{http_code}" http://$LMS_HOST/heartbeat)
+        if [ "$status_code" -ne 200 ]; then
+          echo "Error: LMS Heartbeat endpoint returned status code $status_code"
+          exit 1  # Exit with non-zero code to fail the workflow
+        else
+          echo "Heartbeat endpoint returned status code 200"
+        fi
+      shell: bash
+
+    - name: Run integration tests
+      if: ${{ inputs.shell_file_to_run }}
+      run: |
+        echo "Creating isolated venv to run the tests"
+        python -m venv .venv
+        if [ ! -d ".venv" ]; then
+          echo "Virtual environment creation failed"
+          exit 1
+        fi
```

Comment:
Has the venv creation failed before? I can't think of why it could fail, but there may be edge cases I haven't considered. 

Also, should we create the environment before installing any dependencies like tutor?

Reply:
You're right. I think I initially added that check to ensure there were no issues with the creation, and then I forgot to remove it, but the step really never failed

Thanks!